flake.nix

```diff
@@ -1,26 +1,34 @@
 {
   description = "The purely functional package manager";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    utils.url = "github:numtide/flake-utils";
+  };
 
-  outputs = { self, nixpkgs }: let
-    systems = [ "x86_64-linux" "i686-linux" "aarch64-linux" ];
-    forAllSystems = f: nixpkgs.lib.genAttrs systems (system: f system);
-  in {
-    bundlers = {
-      nix-bundle = { program, system }: let
-        nixpkgs' = nixpkgs.legacyPackages.${system};
-        nix-bundle = import self { nixpkgs = nixpkgs'; };
-        script = nixpkgs'.writeScript "startup" ''
-          #!/bin/sh
-          .${nix-bundle.nix-user-chroot}/bin/nix-user-chroot -n ./nix -- ${program} "$@"
-        '';
-      in nix-bundle.makebootstrap {
-        targets = [ script ];
-        startup = ".${builtins.unsafeDiscardStringContext script} '\"$@\"'";
+  outputs =
+    inputs:
+    let
+      inherit (inputs.nixpkgs) lib;
+    in
+    inputs.utils.lib.eachDefaultSystem (system: {
+      bundlers = {
+        default = inputs.self.bundlers.${system}.nix-bundle;
+        nix-bundle =
+          drv:
+          let
+            program = lib.getExe drv;
```

Comment:
Uses `meta.mainProgram`, the new api doesn't have a way to pass what program path should be used